tests/wat/if.wat

```diff
@@ -0,0 +1,64 @@
+(module
+    (import "wasi_unstable" "fd_write" (func $fd_write (param i32 i32 i32 i32) (result i32)))
+    (import "wasi_unstable" "proc_exit" (func $proc_exit (param i32)))
+    
+    (memory 1)
+    (export "memory" (memory 0))
```

Comment:
Why are these memory numbers different?

Reply:
I had to dig into this a bit. I'm with you on this looking super confusing. If this were to act like other S-expressions, it would get unwrapped as:
```
(memory 1)
(memory 0)
(export "memory")
```

I think the reason this works differently is that the `export` instruction takes two operands:
![image](https://user-images.githubusercontent.com/8700322/125094337-2b8dc580-e0a1-11eb-9883-8f6fa2d983ff.png)

So the second `(memory 0)` is an operand, not an instruction, and in that context, it acts as a lookup that resolves to the memory at index 0.

This blog post shows a few other syntaxes: https://blog.scottlogic.com/2018/04/26/webassembly-by-hand.html

`wat2wasm` ends up compiling them to what is already here.